expression/builtin_string.go

```diff
@@ -2393,7 +2393,7 @@ func (b *builtinFindInSetSig) evalInt(row chunk.Row) (int64, bool, error) {
 	}
 
 	for i, strInSet := range strings.Split(strlist, ",") {
-		if str == strInSet {
+		if collate.GetCollator(b.collation).Compare(str, strInSet, collate.NewCollatorOption(0)) == 0 {
```

Comment:
Do we need a default option to avoid small struct allocation?

Reply:
I think we can remove the length option...even the option itself... but better in another PR.

Here the collator can be constructed on initialization of `b`.

Reply:
I think we can use the pointer, then we can pass `nil` in directly.